datastore/prop_test.go

```diff
@@ -545,3 +545,80 @@ func TestNilKeyIsStored(t *testing.T) {
 		t.Errorf("I field was not zero")
 	}
 }
+
+func TestSaveStructOmitEmpty(t *testing.T) {
+	expectedPropNames := []string{"EmptyValue", "NonEmptyValue", "OmitEmptyWithValue"}
+
+	assert := func(src interface{}) {
+		t.Helper()
+		if props, err := SaveStruct(src); err != nil {
+			t.Fatal(err)
+		} else if len(props) != len(expectedPropNames) {
+			t.Errorf("Expected %v properties, got %v", len(expectedPropNames), len(props))
+		} else {
+			expected := make([]string, len(expectedPropNames))
+			copy(expected, expectedPropNames)
+		PROPS:
```

Comment:
I think it would be more readable to sort `expectedPropNames` and `props` and then compare with a single loop.

```
sort.Slice(props, func(i, j int) bool {
	return props[i].Name < props[j].Name
})

for i, _ := range props {
	if props[i].Name != expectedPropNames[i] {
		t.Errorf("Unexpected property: %v", props[i].Name)
	}
}
```